CONTRIBUTING.md

```diff
@@ -0,0 +1,32 @@
+Esri welcomes contributions from anyone and everyone. Please see our [guidelines for contributing](https://github.com/esri/contributing).
+
+### Before filing an issue
+
+Please take a look at [previous issues](https://github.com/Esri/arcgis-python-api/issues?q=is%3Aissue+is%3Aclosed) that resolve common problems.
+
+If you're just looking for help, you'll probably attract the most eyes if you post in the [GeoNet Forums](https://community.esri.com/groups/arcgis-python-api/) or the [GIS Stackexchange](https://gis.stackexchange.com/questions/tagged/arcgis-python-api).
+
+If you think you're encountering a new bug, please feel free to log an [issue](https://github.com/Esri/arcgis-python-api/issues/new) and include the steps to reproduce the problem.
+
+**Please include the following in your issue:**
+* Your conda or pip environment (run `conda list` or `pip freeze` and copy/paste the output into the issue)
+* Any code (or notebook) you executed that made you run into this issue
+
+# Updating errors in existing notebooks
+
+Do you see an error in an existing notebook? Is there a typo, a misnamed variable, and do you know how to fix it?
+* Make the fix in the notebook
+* If you made code changes, re-run the appropriate cells in the notebook (Don't re-run every cell in the notebook unless necessary)
+* Make a copy of the `dev_site_current` branch on your local fork
+* Add the updated notebook to your branch on your fork
+* [Open a pull request](https://help.github.com/articles/creating-a-pull-request-from-a-fork/) from your fork's branch to this repository's `dev_site_current` branch
+
+# Adding new sample notebooks
+
+Do you have a notebook that shows off functionality of the API? Do you want to see that notebook released with the next version of the ArcGIS API for Python?
+* Create a new notebook, make sure it runs consistently on its own on a few systems
+    * Make sure the notebook filename is in an `all_lowercase_underscore_delimiter_convention.ipynb`
+* Make a copy of the `dev_site_next_release` branch on your local fork
```

Comment:
Why do you recommend "next_release" branch instead of "current" branch for new notebooks?

Reply:
This one could go either way, but it seems to me that new notebooks should be released when we do a software release. They get more visibility since we can talk about them in a blog post, whereas getting published immediately to the developer's website might make them get lost in the shuffle.

I think we can take notebooks on a case-by-case basis of when to release them, but we should outline some default behavior.